### PR TITLE
feat #8887: fixing question order on the attendee overview

### LINKED ISSUE #8887: Sort form fields on the overview page in the same way as on the questionnaire
When making orders the overview page shows all questions that have been filled in on the previous page. These questions are in a different order than the attendee form questions. 

Please ensure the overview pages shows the questions in the same order as the attendee form.

Steps:
1. Get a ticket
2. Fill in the form
3. See the overview page

### PR DESCRIPTION
**Pull request by Bekka592**
Fixes #8887 

The overview pages now shows the custom questions in the same order as the attendee form generator and the attendee form. This also works when the user edited the order of questions.

Here an example with some questions and the question "Github" pulled up to the very top, as well as the question "Work Phone" ordered just below "Phone" (else, all questions are ordered alphabetically - as per default):

Screenshot_20230704_161213

Screenshot_20230704_161234

Screenshot_20230704_161305

Screenshot_20230704_161318

Screenshot_20230704_161350

I also adjusted the order of questions in ```custom-form.js```, just so they are uniform with the question order in the server code. This might otherwise lead to weird bugs later on (however, right now this change makes no difference).

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by mariobehling — APPROVED** [review]

**Review by mariobehling — DISMISSED** [review]

**Comment by cweitat** [comment]
@mariobehling first name and last name are separated. should we proceed with this for now?

File: app/components/forms/orders/attendee-list.js [hunk 1]
```diff
@@ -74,8 +74,9 @@ export default class AttendeeList extends Component {
         field.transName = field.name;
       }
 
-      return field.form;
-    });
+      return true;
```
Comment (cweitat) [inline app/components/forms/orders/attendee-list.js:77]: i think this line of code is redundant. please check thanks!

**Review by cweitat — CHANGES_REQUESTED** [review]

File: app/components/forms/orders/attendee-list.js (hunk 1 above)
Comment (norbusan) [reply app/components/forms/orders/attendee-list.js:77]: But something needs to be returned. I am more concerned about why `true`? And how the sorting works.

Comment (Bekka592) [reply app/components/forms/orders/attendee-list.js]: The sorting process works in analogue to the sorting in the `allFields` method of `order-form.js`. That's also where I got that filtering from, even though the `return true` was pretty useless here. However, as Norbert mentioned, something needs to be retuned. I therefore now switched that `filer` to a `map` function. This returns the `field` objects that got modified in the function, in order to then have them accessible for the sorting. What do you think of that solution?

Comment (cweitat) [reply app/components/forms/orders/attendee-list.js]: This is GTG

**Pull request merged**